tests/unit/components/common/bs-form-test.js

```diff
@@ -0,0 +1,59 @@
+import { setupTest, test } from 'ember-qunit';
+import { reject, resolve } from 'rsvp';
+import { module } from 'qunit';
+
+module('Unit | Component | bs-form', function(hooks) {
+  setupTest(hooks);
+
+  test('calling form.validate returns success if validations pass', function(assert) {
+    assert.expect(1);
+
+    let Factory = this.owner.factoryFor('component:bs-form');
+    let subject = Factory.create({ hasValidator: true, model: {}, validate: function() { return resolve(); } });
+
+    let validatedResult = subject.actions.validate.apply(subject);
```

Comment:
Doing this got a bit ugly, but I wanted to test the promises we were returning to ensure I did them right 😉

Reply:
LGTM! :) Maybe just tweak the name of the test, to express that it returns a resolving promise?